.github/workflows/tag-and-release.yml

```diff
@@ -0,0 +1,107 @@
+name: Tag & Release
+on:
+  workflow_dispatch:
+jobs:
+  tag:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+      - name: Get Version
+        run: |
+          export VERSION=$(sed -n -e 's/.*QVersionNumber::fromString(\"\(.*\)\");/\1/p' src/window.cpp)
+          git tag "v$VERSION"
+          git push origin "v$VERSION"
+  build-linux-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+      - uses: jurplel/install-qt-action@37e649dae477dba3481cbec4788e5b3ca9d6e170
+        with:
+          cache: true
+      - name: build
+        run: |
+          qmake
+          make
+          tar -czvf minikube-gui-linux.tar.gz ./minikube
+      - uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce
+        with:
+          name: minikube-gui-linux-release
+          path: ./minikube-gui-linux.tar.gz
+  build-macos-release:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+      - uses: jurplel/install-qt-action@37e649dae477dba3481cbec4788e5b3ca9d6e170
+        with:
+          cache: true
+      - name: build
+        run: |
+          qmake
+          make
+      - name: package
+        run: |
+          macdeployqt ./minikube.app -qmldir=. -verbose=1 -dmg
```

Comment:
not needed for this PR but consider moving these things to a Make Target so it can be called form different jobs and when we need to update them we jsut update one